inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/InApp.kt

```diff
@@ -75,12 +74,13 @@ internal class InApp(
     @RestrictTo(RestrictTo.Scope.LIBRARY)
     override fun getHostAppContext() = context
 
-    override fun closeMessage() {
+    override fun closeMessage(clearQueuedCampaigns: Boolean) {
         val id = displayManager.removeMessage(getRegisteredActivity())
 
-        if (id != null) {
-            // Remove message from ReadyForDisplayMessageRepository.
-            val message = ReadyForDisplayMessageRepository.instance().removeMessage(id as String)
+        if (clearQueuedCampaigns) {
+            ReadyForDisplayMessageRepository.instance().clearMessages(clearQueuedCampaigns)
```

Comment:
can this method be called with `false`. Does that even make sense?

Reply:
Inside `closeMessage()` API, no. But yes, it is possible to call the method with `false` input.
In that case, the campaigns' `timesClosed` will not be incremented.

Reply:
yeah but is `clearMessages(false)` used anywhere?

Reply:
yes, in `SessionManager#onSessionUpdate()`.